site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,115 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TextField from "@mui/material/TextField";
+import { useState, type FC } from "react";
+
+type TimeUnit = "days" | "hours";
+
+// Value should be in milliseconds or undefined. Undefined means no value.
+type DurationValue = number | undefined;
+
+type DurationFieldProps = {
+  label: string;
+  value: DurationValue;
+  onChange: (value: DurationValue) => void;
+};
+
+export const DurationField: FC<DurationFieldProps> = (props) => {
+  const { label, value, onChange } = props;
+  const [timeUnit, setTimeUnit] = useState<TimeUnit>(() => {
+    if (!value) {
+      return "hours";
+    }
+
+    return Number.isInteger(durationToDays(value)) ? "days" : "hours";
+  });
+
+  return (
+    <div
+      css={{
+        display: "flex",
+        gap: 8,
+      }}
+    >
+      <TextField
+        css={{ maxWidth: 160 }}
+        label={label}
+        value={
+          !value
+            ? ""
+            : timeUnit === "hours"
+              ? durationToHours(value)
+              : durationToDays(value)
+        }
```

Comment:
I'm not fully clear on how the `value` prop for `TextField` works when you give the component `inputProps` with a type of `number`

Could the empty string be replaced with `0`, and if so, do you think it makes sense to make that the default value for `DurationField`'s `value` prop, so that you don't have to juggle `undefined` and "zero-ish" values?